ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/bootstrap/tasks/partials/ubuntu1604.yml

```diff
@@ -0,0 +1,19 @@
+---
+
+#
+#  ubuntu 16.04
+#
+
+- name: check for python
+  raw: stat /usr/bin/python
+  failed_when: has_python.rc > 1
+  register: has_python
+
+- name: check for aptitude
+  raw: stat /usr/bin/aptitude
+  failed_when: has_aptitude.rc > 1
+  register: has_aptitude
+
+- name: install python and aptitude
+  when: has_python.rc == 1 or has_aptitude.rc == 1
+  raw: apt-get update && apt install -y python-minimal aptitude
```

Comment:
Unless something has changes ansible has generally been able to fall back to `apt` instead of `aptitude` if it's not available which makes this unnecesary

Reply:
`aptitude` is considered more efficient for bulk package upgrades

Reply:
Maybe, but I don't think we should necessarily mandate it. Do we know how much of a difference it has? I'm generally not in favour of putting stuff on the machines that isn't either necessary or provides a proven benefit to us.

Reply:
the playbooks ran noticeably faster with aptitude installed when I did some tests